diff --git a/src/fr/anssi/smartpgp/SmartPGPApplet.java b/src/fr/anssi/smartpgp/SmartPGPApplet.java
index 6be8d3f..32108cc 100644
--- a/src/fr/anssi/smartpgp/SmartPGPApplet.java
+++ b/src/fr/anssi/smartpgp/SmartPGPApplet.java
@@ -1333,10 +1333,12 @@ public final class SmartPGPApplet extends Applet {
             return;
         }
 
+/*
         if(data.admin_pin.getTriesRemaining() <= 0) {
             data.isTerminated = true;
             return;
         }
+*/
 
         assertAdmin();
 
