diff --git a/src/fr/anssi/smartpgp/SmartPGPApplet.java b/src/fr/anssi/smartpgp/SmartPGPApplet.java
index d8b4257..1b11006 100644
--- a/src/fr/anssi/smartpgp/SmartPGPApplet.java
+++ b/src/fr/anssi/smartpgp/SmartPGPApplet.java
@@ -1387,10 +1387,12 @@ public final class SmartPGPApplet extends Applet {
             return;
         }
 
+/*
         if(data.admin_pin.getTriesRemaining() <= 0) {
             data.isTerminated = true;
             return;
         }
+*/
 
         assertAdmin();
 
