diff --git a/src/fr/anssi/smartpgp/Constants.java b/src/fr/anssi/smartpgp/Constants.java
index 6a66b73..fb9d1d1 100644
--- a/src/fr/anssi/smartpgp/Constants.java
+++ b/src/fr/anssi/smartpgp/Constants.java
@@ -149,7 +149,7 @@ public final class Constants {
         (byte)0x73, /* card capabilities */
         (byte)0xC0, /* 1st byte: "methods supported" see ISO 7816-4 */
         (byte)0x01, /* 2nd byte: "data coding byte" idem */
-        (byte)0x80, /* 3rd byte: command chaining (not extended length by default as all readers do not support them...) */
+        (byte)0xC0, /* 3rd byte: command chaining + extended length */
 
         (byte)0x05, /* status indicator byte : operational state */
         (byte)0x90, /* SW1 */
